x-pack/plugins/apm/server/feature.ts

```diff
@@ -6,33 +6,34 @@
  */
 
 import { i18n } from '@kbn/i18n';
+import { SubFeaturePrivilegeGroupType } from '../../features/common';
 import { LicenseType } from '../../licensing/common/types';
-import { AlertType } from '../common/alert_types';
+import { AlertType, APM_SERVER_FEATURE_ID } from '../common/alert_types';
 import { DEFAULT_APP_CATEGORIES } from '../../../../src/core/server';
 import {
   LicensingPluginSetup,
   LicensingApiRequestHandlerContext,
 } from '../../licensing/server';
 
 export const APM_FEATURE = {
-  id: 'apm',
+  id: APM_SERVER_FEATURE_ID,
   name: i18n.translate('xpack.apm.featureRegistry.apmFeatureName', {
     defaultMessage: 'APM and User Experience',
   }),
   order: 900,
   category: DEFAULT_APP_CATEGORIES.observability,
-  app: ['apm', 'ux', 'kibana'],
-  catalogue: ['apm'],
+  app: [APM_SERVER_FEATURE_ID, 'ux', 'kibana'],
+  catalogue: [APM_SERVER_FEATURE_ID],
   management: {
     insightsAndAlerting: ['triggersActions'],
   },
   alerting: Object.values(AlertType),
   // see x-pack/plugins/features/common/feature_kibana_privileges.ts
   privileges: {
     all: {
-      app: ['apm', 'ux', 'kibana'],
-      api: ['apm', 'apm_write'],
-      catalogue: ['apm'],
+      app: [APM_SERVER_FEATURE_ID, 'ux', 'kibana'],
+      api: [APM_SERVER_FEATURE_ID, 'apm_write', 'rac'],
+      catalogue: [APM_SERVER_FEATURE_ID],
```

Comment:
Can't comment directly on the line but on line 46, the feature privilege should be changed to `read` if you are adding the subfeature privilege section, so

```
alerting {
  rule: {
    all: Object.values(AlertType),
  },
  alert: {
    read: Object.values(AlertType),
  }
}
```

Otherwise if you give `all` alerting privilege but uncheck the subfeature privilege checkbox for alerts, it won't actually de-escalate the `alerts` privilege.